Embedded version makes development difficult

I have used `go run` during development, but now it doesn't work:

```console
$ cd cmd/turso/
$ go run main.go
../../internal/cmd/root.go:11:12: pattern version.txt: no matching files found
```

Let's use the build tag solution for conditionally including version information depending on whether it's a development or production build:

https://blog.carlmjohnson.net/post/2021/how-to-use-go-embed/

--- a/.goreleaser.yaml
+++ b/.goreleaser.yaml
@@ -6,7 +6,7 @@ before:
     # You may remove this if you don't use go modules.
     - go mod tidy
     # you may remove this if you don't need go generate
-    - go generate ./...
+    - go generate --tags=prod ./...
     - ./script/completions.sh
 builds:
   - env:
@@ -17,6 +17,8 @@ builds:
       - darwin
     binary: turso
     main: ./cmd/turso
+    flags:
+      - -tags=prod
 
 release:
   github:

--- a/internal/cmd/root.go
+++ b/internal/cmd/root.go
@@ -1,15 +1,10 @@
 package cmd
 
 import (
-	_ "embed"
 	"os"
 
 	"github.com/spf13/cobra"
 )
-
-//go:generate sh -c "printf %s $(../../script/version.sh) > version.txt"
-//go:embed version.txt
-var version string
 
 var rootCmd = &cobra.Command{
 	Use:     "turso",

--- a/internal/cmd/version_dev.go
+++ b/internal/cmd/version_dev.go
@@ -1,0 +1,7 @@
+// version_dev.go
+//go:build !prod
+// +build !prod
+
+package cmd
+
+var version string = "dev"

--- a/internal/cmd/version_prod.go
+++ b/internal/cmd/version_prod.go
@@ -1,0 +1,13 @@
+// version_prod.go
+//go:build prod
+// +build prod
+
+package cmd
+
+import (
+	_ "embed"
+)
+
+//go:generate sh -c "printf %s $(../../script/version.sh) > version.txt"
+//go:embed version.txt
+var version string

